server/src/test/java/org/opensearch/search/approximate/ApproximatePointRangeQueryTests.java

```diff
@@ -29,54 +35,277 @@
+                { NumericType.DOUBLE },
+                { NumericType.UNSIGNED_LONG } }
+        );
+    }
+
+    enum NumericType {
+        INT("int_field", ApproximatePointRangeQuery.INT_FORMAT) {
+            @Override
+            byte[] encode(Number value) {
+                return IntPoint.pack(new int[] { value.intValue() }).bytes;
+            }
+
+            @Override
+            void addField(Document doc, String fieldName, Number value) {
+                doc.add(new IntPoint(fieldName, value.intValue()));
+            }
+
+            @Override
+            void addDocValuesField(Document doc, String fieldName, Number value) {
+                doc.add(new NumericDocValuesField(fieldName, value.intValue()));
+            }
+
+            @Override
+            Query rangeQuery(String fieldName, Number lower, Number upper) {
+                return IntPoint.newRangeQuery(fieldName, lower.intValue(), upper.intValue());
+            }
+
+            @Override
+            SortField.Type getSortFieldType() {
+                return SortField.Type.INT;
+            }
+        },
+        LONG("long_field", ApproximatePointRangeQuery.LONG_FORMAT) {
+            @Override
+            byte[] encode(Number value) {
+                return LongPoint.pack(new long[] { value.longValue() }).bytes;
+            }
+
+            // Add to BKD
+            @Override
+            void addField(Document doc, String fieldName, Number value) {
+                doc.add(new LongPoint(fieldName, value.longValue()));
+            }
+
+            // Add DocValues
+            @Override
+            void addDocValuesField(Document doc, String fieldName, Number value) {
+                doc.add(new NumericDocValuesField(fieldName, value.longValue()));
+            }
+
+            @Override
+            Query rangeQuery(String fieldName, Number lower, Number upper) {
+                return LongPoint.newRangeQuery(fieldName, lower.longValue(), upper.longValue());
+            }
+
+            @Override
+            SortField.Type getSortFieldType() {
+                return SortField.Type.LONG;
+            }
+        },
+        HALF_FLOAT("half_float_field", ApproximatePointRangeQuery.HALF_FLOAT_FORMAT) {
+            @Override
+            byte[] encode(Number value) {
+                byte[] bytes = new byte[HalfFloatPoint.BYTES];
+                HalfFloatPoint.encodeDimension(value.floatValue(), bytes, 0);
+                return bytes;
+            }
+
+            @Override
+            void addField(Document doc, String fieldName, Number value) {
+                doc.add(new HalfFloatPoint(fieldName, value.floatValue()));
+            }
+
+            @Override
+            void addDocValuesField(Document doc, String fieldName, Number value) {
+                doc.add(new NumericDocValuesField(fieldName + "_sort", value.longValue()));
+            }
+
+            @Override
+            String getSortFieldName() {
```

Comment:
Why do the sort fields for `half_float` and `unsigned_long` need to be overrided?

Reply:
This is because when testing using with Lucene `Sort sort = new Sort()` it expects a `SortField.Type`, using directly `half_float` we see error as 
```
Field half_float_field is indexed with 2 bytes per dimension, but org.apache.lucene.search.comparators.LongComparator@7da5f124 expected 8`
```
Hence to test the `half_float` or `unsigned_long` I'm using `NumericDocValuesField` which is indexed as different field other than the one that is added to BKD.

For OpenSearch custom comparators is used to handle this which was added is https://github.com/opensearch-project/OpenSearch/pull/11024 which AFIAK
- Read the 2-byte point values.
- Convert them properly.
- Compare them as floats.

Reply:
Essentially these tests ensure the order, return of documents, hits, values do not change with the approximation code. If approximation is setting or flag we can directly test with OpenSearch with flag enable and disable, since this is not the case the tests run directly with Lucene and compare with OpenSearch.